firmware/src/engine/share.cpp

```
@@ -0,0 +1,387 @@
+    return true;
+  }
+
+  bool accept(const char* text) {
+    const size_t length = std::strlen(text);
+    if (std::strncmp(at_, text, length) != 0) return false;
+    at_ += length;
+    return true;
+  }
+
+  // Digits only, at least one; leading zeros load and re-encode canonically (D-039).
+  bool read_int(int min, int max, int& out) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)